crates/ra_ide/src/syntax_highlighting/tests.rs

```diff
@@ -325,6 +325,43 @@ impl Foo {
     )
 }
 
+#[test]
+fn test_recursive_macro_highlighting() {
+    check_highlighting(
+        r##"
+macro_rules! format {
+    ($($arg:tt)*) => {{
+        let res = $crate::fmt::format($crate::format_args!($($arg)*));
+        res
+    }}
+}
+macro_rules! assert_eq {
+    ($left:expr, $right:expr) => ({
+        match (&$left, &$right) {
+            (left_val, right_val) => {
+                if !(*left_val == *right_val) {
+                    // The reborrows below are intentional. Without them, the stack slot for the
+                    // borrow is initialized even before the values are compared, leading to a
+                    // noticeable slow down.
+                    panic!(r#"assertion failed: `(left == right)`
+  left: `{:?}`,
+ right: `{:?}`"#, &*left_val, &*right_val)
+                }
+            }
+        }
+    });
+}
+
+fn main() {
+    assert_eq!("1", format!("{}", 1));
+}
+"##
```

Comment:
Let's minimize this test, and maybe fold it into the main highlighting test?  I'd love to avoid ballooning the amount of test data :)

Reply:
I integrated a minimal example into the main highlighting test